Source/NSButton.m

```diff
@@ -543,4 +547,43 @@ - (NSSound *) sound
   return [_cell sound];
 }
 
+- (void) mouseDown: (NSEvent *)event
+{
+  [super mouseDown: event];
+
+  if ([[self superview] isKindOfClass: [NSMatrix class]] == NO)
+  {
+    NSImage *i = [[self cell] image];
+    BOOL isRadio = (i == [NSImage imageNamed: @"NSRadioButton"]);
+  
+    if (isRadio)
+      {
+        NSView *sv = [self superview];
+        NSArray *subviews = [sv subviews];
+        
+        FOR_IN(NSView*, v, subviews)
+          {
+          if ([v isKindOfClass: [NSButton class]])
+            {
+              NSButton *b = (NSButton *)v;
+              NSImage *image = [[b cell] image];
+              BOOL flag = (image == [NSImage imageNamed: @"NSRadioButton"]);
+              
+              if (flag)
+                {
+                  if ([self action] == [b action] && b != self)
+                    {
+                      if ([self state] == NSOnState)
```

Comment:
This condition should be checked outside of the loop, probably right at the start of the method

Reply:
Fixed.